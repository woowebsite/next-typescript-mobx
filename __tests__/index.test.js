/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/index.tsx'

describe('With Enzyme', () => {
  it('App has link About', () => {
    const app = shallow(<App />)

    expect(app.find('p').find('a').text()).toEqual('About')
  })
})

describe('With Snapshot Testing', () => {
  it('App has link About', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
