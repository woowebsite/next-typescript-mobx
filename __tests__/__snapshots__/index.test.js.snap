// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App has link About 1`] = `
<div>
  <header>
    <nav>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
       |
       
      <a
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </nav>
  </header>
  <h1>
    Hello Next.js ðŸ‘‹
  </h1>
  <p>
    <a
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
  </p>
  <footer>
    I'm here to stay
  </footer>
</div>
`;
